Orm/Xtensive.Orm/Orm/Session.SystemLogic.cs

```diff
@@ -11,17 +11,26 @@ namespace Xtensive.Orm
 {
   public partial class Session
   {
+    public readonly struct SystemLogicOnlyRegionScope : IDisposable
```

Comment:
> Do you have any usage of `DirectSessionAccessor.OpenSystemLogicOnlyRegion` method?

Yes, we do

Reply:
Got it

Reply:
It is reasonable to restrict access to the scope constructor to keep creations within project

Reply:
> roll back to `IDisposable` in the `DirectSessionAccessor`.

returning `IDisposable` requires heap allocation. That was the reason of this PR to avoid unnecessary allocations.

Reply:
I know, this is why I no longer require you to make type internal. I'm talking about the type constructor, there is no reason for it to be used outside project.

Reply:
Changed `SystemLogicOnlyRegionScope()` ctor to be `internal`